stdlib/internal/promql/join.md

```diff
@@ -0,0 +1,91 @@
+### join
+
+`join` merges two input streams into a single output stream.
+Records that have the same group key and `_time` values will be joined in the output.
+The output groups will be the same as the input groups.
+
+`join` has the following properties:
+
+| Name    | Type     | Description |
+| ------- | -------- | ----------- |
+| tables  | object   | Specifies the two table streams to join. |
+| fn      | function | Defines the function that joins the rows of each table. The return value is an object which defines the output record structure. |
+
+Ex.
+```
+import "internal/promql"
+
+left = from(bucket: "my-bucket")
+    |> range(start: -1h)
+    |> filter(fn: (r) => r._field == "A")
+    |> drop(columns: ["_field"])
+    |> rename(columns: {_value: "a"})
+
+right = from(bucket: "my-bucket")
+    |> range(start: -1h)
+    |> filter(fn: (r) => r._field == "B")
+    |> drop(columns: ["_field"])
+    |> rename(columns: {_value: "b"})
+
+promql.join(tables: {left, right}, fn: (t) => {
+    return {t.left with b: t.right.b}
+}
+```
+
+`join` assumes that each input table is sorted by `_time`.
+The planner will ensure this by inserting a `sort` procedure after each one of join's immediate predecessors.
+`join` will only call `fn` on rows that pass the join condition.
+
+Here are some examples of compiling PromQL queries into their equivalent Flux forms involving `promql.join`.
+
+```
+left_metric + right_metric
```

Comment:
Ah, PromQL syntax, so concise, so beautiful.

Reply:
Just to clarify, the full transpilation of this query will require additional postprocessing, at least also dropping the metric name (`_field`) column here. In other cases (like the example below that uses `ignoring()`), the ignored tags also need to be dropped from the final result, etc.. So the `join()` will always just be a fragment (though crucial part) of what's needed for the transpilation.

Reply:
I forgot to drop the `_field` and ignored columns in these examples. I will update.